nativeauthenticator/handlers.py

```diff
@@ -266,12 +273,24 @@ async def get(self):
     @web.authenticated
     async def post(self):
         user = await self.get_current_user()
-        new_password = self.get_body_argument("password", strip=False)
-        success = self.authenticator.change_password(user.name, new_password)
+        old_password = self.get_body_argument("old_password", strip=False)
+        new_password = self.get_body_argument("new_password", strip=False)
+        confirmation = self.get_body_argument("con_password", strip=False)
+
+        correctpw = self.authenticator.get_user(user.name).is_valid_password(
+            old_password
+        )
+
+        success = self.authenticator.user_change_password(
+            user.name, old_password, new_password, confirmation
+        )
```

Comment:
If `(user_)change_password()` is not doing the validation check anymore as requested, we will need to do the validation check before attempting the password change since it is now not only for the precise error message (which I think is important) but also to make sure the password only gets changed if the correct current password was provided.